In [218]:
import urllib.request
from bs4 import BeautifulSoup
import re
import json
import time

In [177]:
def generate_classification_links():
    sourcelink = 'http://www.mftd.org/index.php?action=atu&act=select&atu='
    file = open('generate_classification_links.txt', 'w', encoding = 'utf-8')
    links = []
    for i in range (1, 2400):
        links.append(sourcelink + str(i) + '\n')
    js = json.dumps(links, ensure_ascii = False)
    file.write(js)

In [178]:
generate_classification_links()

In [179]:
def open_link(link):
    a = urllib.request.urlopen(link)
    file = a.read()
    return file

In [224]:
def database_with_ids():
    time.sleep(20)
    file_r = open('generate_classification_links.txt', 'r', encoding = 'utf-8')
    file_r = json.loads(file_r.read())
    file_w = open('database_with_ids.txt', 'w', encoding = 'utf-8')
    dat = {}
    COUNT = 0
    for line in (file_r):
        everything = []
        soup = boil_some_soup(line)
        try:
            number, name = number_name(soup)
            hier = get_hier(soup)
            ids = get_ex_ids(soup)
            everything.append(name)
            everything.append(hier)
            everything.append(ids)
            dat[number] = everything
        except ValueError:
            print(soup.h2.text)    
        except AttributeError:
            print(str(COUNT) + ' missin ' + line)
            COUNT +=1
    js = json.dumps(dat, ensure_ascii = False)
    file_w.write(js)

In [225]:
database_with_ids()

0 missin http://www.mftd.org/index.php?action=atu&act=select&atu=6

1 missin http://www.mftd.org/index.php?action=atu&act=select&atu=10

2 missin http://www.mftd.org/index.php?action=atu&act=select&atu=11

3 missin http://www.mftd.org/index.php?action=atu&act=select&atu=12

4 missin http://www.mftd.org/index.php?action=atu&act=select&atu=13

5 missin http://www.mftd.org/index.php?action=atu&act=select&atu=14

6 missin http://www.mftd.org/index.php?action=atu&act=select&atu=16

7 missin http://www.mftd.org/index.php?action=atu&act=select&atu=17

8 missin http://www.mftd.org/index.php?action=atu&act=select&atu=18

9 missin http://www.mftd.org/index.php?action=atu&act=select&atu=19

10 missin http://www.mftd.org/index.php?action=atu&act=select&atu=22

11 missin http://www.mftd.org/index.php?action=atu&act=select&atu=23

12 missin http://www.mftd.org/index.php?action=atu&act=select&atu=24

13 missin http://www.mftd.org/index.php?action=atu&act=select&atu=25

14 missin http://www.mftd.org/i

116 missin http://www.mftd.org/index.php?action=atu&act=select&atu=265

117 missin http://www.mftd.org/index.php?action=atu&act=select&atu=266

118 missin http://www.mftd.org/index.php?action=atu&act=select&atu=267

119 missin http://www.mftd.org/index.php?action=atu&act=select&atu=268

120 missin http://www.mftd.org/index.php?action=atu&act=select&atu=269

121 missin http://www.mftd.org/index.php?action=atu&act=select&atu=270

122 missin http://www.mftd.org/index.php?action=atu&act=select&atu=271

123 missin http://www.mftd.org/index.php?action=atu&act=select&atu=272

124 missin http://www.mftd.org/index.php?action=atu&act=select&atu=273

125 missin http://www.mftd.org/index.php?action=atu&act=select&atu=274

126 missin http://www.mftd.org/index.php?action=atu&act=select&atu=276

127 missin http://www.mftd.org/index.php?action=atu&act=select&atu=279

128 missin http://www.mftd.org/index.php?action=atu&act=select&atu=281

129 missin http://www.mftd.org/index.php?action=atu&act=select&a

230 missin http://www.mftd.org/index.php?action=atu&act=select&atu=464

231 missin http://www.mftd.org/index.php?action=atu&act=select&atu=467

232 missin http://www.mftd.org/index.php?action=atu&act=select&atu=469

233 missin http://www.mftd.org/index.php?action=atu&act=select&atu=472

234 missin http://www.mftd.org/index.php?action=atu&act=select&atu=473

235 missin http://www.mftd.org/index.php?action=atu&act=select&atu=474

236 missin http://www.mftd.org/index.php?action=atu&act=select&atu=477

237 missin http://www.mftd.org/index.php?action=atu&act=select&atu=478

238 missin http://www.mftd.org/index.php?action=atu&act=select&atu=479

239 missin http://www.mftd.org/index.php?action=atu&act=select&atu=481

240 missin http://www.mftd.org/index.php?action=atu&act=select&atu=482

241 missin http://www.mftd.org/index.php?action=atu&act=select&atu=483

242 missin http://www.mftd.org/index.php?action=atu&act=select&atu=484

243 missin http://www.mftd.org/index.php?action=atu&act=select&a

344 missin http://www.mftd.org/index.php?action=atu&act=select&atu=649

345 missin http://www.mftd.org/index.php?action=atu&act=select&atu=651

346 missin http://www.mftd.org/index.php?action=atu&act=select&atu=656

347 missin http://www.mftd.org/index.php?action=atu&act=select&atu=657

348 missin http://www.mftd.org/index.php?action=atu&act=select&atu=658

349 missin http://www.mftd.org/index.php?action=atu&act=select&atu=659

350 missin http://www.mftd.org/index.php?action=atu&act=select&atu=661

351 missin http://www.mftd.org/index.php?action=atu&act=select&atu=662

352 missin http://www.mftd.org/index.php?action=atu&act=select&atu=663

353 missin http://www.mftd.org/index.php?action=atu&act=select&atu=664

354 missin http://www.mftd.org/index.php?action=atu&act=select&atu=666

355 missin http://www.mftd.org/index.php?action=atu&act=select&atu=668

356 missin http://www.mftd.org/index.php?action=atu&act=select&atu=669

357 missin http://www.mftd.org/index.php?action=atu&act=select&a

458 missin http://www.mftd.org/index.php?action=atu&act=select&atu=904

459 missin http://www.mftd.org/index.php?action=atu&act=select&atu=905

460 missin http://www.mftd.org/index.php?action=atu&act=select&atu=906

461 missin http://www.mftd.org/index.php?action=atu&act=select&atu=907

462 missin http://www.mftd.org/index.php?action=atu&act=select&atu=908

463 missin http://www.mftd.org/index.php?action=atu&act=select&atu=909

464 missin http://www.mftd.org/index.php?action=atu&act=select&atu=911

465 missin http://www.mftd.org/index.php?action=atu&act=select&atu=912

466 missin http://www.mftd.org/index.php?action=atu&act=select&atu=913

467 missin http://www.mftd.org/index.php?action=atu&act=select&atu=914

468 missin http://www.mftd.org/index.php?action=atu&act=select&atu=917

469 missin http://www.mftd.org/index.php?action=atu&act=select&atu=918

470 missin http://www.mftd.org/index.php?action=atu&act=select&atu=919

925: Tidings Brought to the King: You Said it, not I
471 missin 

570 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1123

571 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1124

572 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1125

573 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1126

574 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1127

575 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1128

576 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1129

577 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1134

578 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1136

579 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1139

580 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1140

581 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1141

582 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1144

583 missin http://www.mftd.org/index.php?action=atu

682 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1370

683 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1371

684 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1374

685 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1375

686 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1376

687 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1379

688 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1388

689 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1389

690 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1390

691 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1392

692 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1394

693 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1395

694 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1396

695 missin http://www.mftd.org/index.php?action=atu

1577: Blind Men Duped into Fighting: Money to be Devided
795 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1578

796 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1580

797 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1581

798 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1582

799 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1583

800 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1584

801 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1585

802 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1588

803 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1593

804 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1594

805 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1595

806 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1596

807 missin http://www.mftd.org/index.php?action=atu&act=select&atu=

907 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1758

908 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1759

909 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1760

910 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1761

911 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1762

912 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1763

913 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1764

914 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1765

915 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1766

916 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1767

917 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1768

918 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1769

919 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1770

920 missin http://www.mftd.org/index.php?action=atu

1018 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1934

1019 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1936

1020 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1937

1021 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1938

1022 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1939

1023 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1941

1024 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1942

1025 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1943

1026 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1944

1027 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1945

1028 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1946

1029 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1947

1030 missin http://www.mftd.org/index.php?action=atu&act=select&atu=1949

1031 missin http://www.mftd.org/index.

1130 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2083

1131 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2084

1132 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2085

1133 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2086

1134 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2087

1135 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2088

1136 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2089

1137 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2090

1138 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2091

1139 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2092

1140 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2093

1141 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2094

1142 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2095

1143 missin http://www.mftd.org/index.

1241 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2194

1242 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2195

1243 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2196

1244 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2197

1245 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2198

1246 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2199

1247 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2202

1248 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2203

1249 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2206

1250 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2207

1251 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2208

1252 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2209

1253 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2210

1254 missin http://www.mftd.org/index.

1353 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2315

1354 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2316

1355 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2317

1356 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2318

1357 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2319

1358 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2321

1359 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2322

1360 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2323

1361 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2324

1362 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2325

1363 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2326

1364 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2327

1365 missin http://www.mftd.org/index.php?action=atu&act=select&atu=2328

1366 missin http://www.mftd.org/index.

In [226]:
def boil_some_soup(link):
    try:
        some_text = open_link(link)
        some_text
        soup = BeautifulSoup(some_text)
    except TimeoutError:
        print('мимо')
    return soup

In [181]:
def number_name(soup):
    name = soup.h2.text
    number, name = re.split(': ', name)
    return (number, name)

In [182]:
def get_hier(soup):
    something = soup.find_all('div')
    something = something[1]
    ps = something.find_all('p')
    hier = []
    for p in range(3):
        p = re.match(r'<p>(.*)<a href="index\.php\?action=atu&amp;act=range&amp;id=(.*)', str(ps[p]))
        try:
            p = p.group()
            p = re.sub(r'<p>(.*)<a href="index\.php\?action=atu&amp;act=range&amp;id=(.*)">(.*)<\/a> ', '', p)
            p = re.sub(r'</p>', '', p)
            hier.append(p)
        except AttributeError:
            break
    return hier

In [183]:
def get_ex_ids(soup):
    table_examples = soup.table
    ids = re.findall(r'<a href="index\.php\?action=story&amp;act=select&amp;id=(.*)">', str(table_examples))
    return ids

в database_with_ids.txt структура такая:
{номер мотива: [название мотива,[1, 2, 3], [id примеров]]}
хранится в джесоне
айтемов получилось чуть меньше тысячи


потерялись:
613
925
969
1070
1153
1183
1453
1577
1624
1738
1812
1836A

если решим их не терять с ними надо все тоже самое сделать –– только как-то по-другому парсить number_name(), потому что моя функция сплитит номер и имя по двоеточию, а у них оно встречается в имени

?как хранить сказки:
{номер мотива: [название мотива,[1, 2, 3], [[пример,шапка],[пример,шапка],...]}